kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt

```diff
@@ -128,10 +128,11 @@ class PolygonShape : Shape(ShapeType.POLYGON) {
         var n = MathUtils.min(num, Settings.maxPolygonVertices)
 
         // Perform welding and copy vertices into local buffer.
-        val ps = if (vecPool != null)
-            vecPool[Settings.maxPolygonVertices]
+        val ps : Array<Vec2?>
+        if (vecPool != null)
+            ps = vecPool[Settings.maxPolygonVertices] as Array<Vec2?>
         else
-            arrayOfNulls<Vec2>(Settings.maxPolygonVertices)
+            ps = arrayOfNulls<Vec2>(Settings.maxPolygonVertices)
```

Comment:
This change shouldn't be required, right?

Reply:
Above: 

Note that the cast on line 133 is necessary for Kotlin to figure out that the actual fix on line 147 is valid

Reply:
Now the declaration and the assignment is split but it looks the same to me and it was compiling before. What happens if you restore the original lines there?